src/Controller/Game/RegionController.php

```diff
@@ -199,6 +190,7 @@ public function buy(Request $request, int $regionId): Response
                 ]);
             }
 
+            $em = $this->getEm();
             $player->setCash($player->getCash() - $regionPrice);
```

Comment:
The responsibility of properly decreasing the players cache now relies on the implementor, you could consider encapsulating cash modifications with certain methods like `$player->spend($regionPrice);`

Reply:
You could also do that for the other changes. For example, if Region is a property of Player, you could implement `moveRegion($newRegion);`. It's about inverting that form of control.